classes/helper.php

```diff
@@ -65,12 +65,13 @@ class helper {
      * @param string $iprestriction optional IP range to restrict access to.
      *
      * @return string the created user key value.
+     * @throws \moodle_exception if user doesn't have permission to create key.
      */
     public static function create_user_key(string $iprestriction = '') : string {
         global $USER;
 
-        if (empty($USER->id)) {
-            throw new \coding_exception('Cannot create a user key when not logged in as a user.');
+        if (!has_capability('tool/pdfpages:createaccesskey', \context_system::instance())) {
+            throw new \moodle_exception('error:permissions:createkey', 'tool_pdfpages');
```

Comment:
you can replace that with require_capability(\context_system::instance()), 'tool/pdfpages:createaccesskey', 'error:permissions:createkey', 'tool_pdfpages')

Reply:
Replaced as suggested